robottelo/hosts.py

```diff
@@ -359,3 +366,121 @@ def patch_os_release_version(self, distro=DISTRO_RHEL7):
         else:
             raise ContentHostError('No distro package available to retrieve release version')
         return self.execute(f"echo '{rh_product_os_releasever}' > /etc/yum/vars/releasever")
+
+
+class Capsule(Host):
+    def restart_services(self):
+        """Restart services, returning True if passed and stdout if not"""
+        result = self.execute('foreman-maintain service restart').status
+        return True if result.status == 0 else result.stdout
+
+    def check_services(self):
+        error_msg = 'Some services are not running'
+        result = self.execute('foreman-maintain service status')
+        if result.status == 0:
+            return True
+        for line in result.stdout.splitlines():
+            if error_msg in line:
+                return line.replace(error_msg, '').strip()
+
+    def installer(self, **extra_kwargs):
+        """General purpose installer"""
+        command = f'satellite-installer --scenario {self.__class__.__name__.lower()}'
+        extras = ' '.join(
+            [f'--{key.replace("_", "-")} {value}' for key, value in extra_kwargs.items()]
+        )
+        return self.execute(f'{command} {extras}')
+
+
+class Satellite(Capsule):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._init_nailgun()
+        self._init_cli()
+        self._init_airgun()
+
+    def _init_nailgun(self):
+        """Import all nailgun entities and wrap them under self.api"""
+        from nailgun.config import ServerConfig
+        from nailgun.entity_mixins import Entity
+        from nailgun import entities
+
+        def inject_config(cls, server_config):
+            """inject a nailgun server config into the init of nailgun entity classes"""
+            import functools
+
+            class DecClass(cls):
+                __init__ = functools.partialmethod(cls.__init__, server_config=server_config)
+
+            return DecClass
+
+        # set the server configuration to point to this satellite
+        self.nailgun_cfg = ServerConfig(
+            auth=(settings.server.admin_username, settings.server.admin_password),
+            url=f'https://{self.hostname}',
+            verify=False,
+        )
+        # add each nailgun entity to self.api, injecting our server config
+        self.api = lambda: None
+        for name, obj in entities.__dict__.items():
+            try:
+                if Entity in obj.mro():
+                    setattr(self.api, name, inject_config(obj, self.nailgun_cfg))
+            except AttributeError:
+                # not everything has an mro method, we don't care about them
+                pass
+
+    def _init_cli(self):
+        """Import all robottelo cli entities and wrap them under self.cli"""
+        import importlib
+        from robottelo.cli.base import Base
+
+        self.cli = lambda: None
+        for file in Path('robottelo/cli/').iterdir():
```

Comment:
Consider using entrypoints to explicitly define what classes should be loaded, instead of everything in the cli directory.